procedures/unit-testing-tracing-analytics.ipf

```diff
@@ -32,4 +48,33 @@ static Function/WAVE GetTotals()
 	return totals
 End
 
+static Function CollectFunctions(WAVE totals, WAVE/T procs, STRUCT CollectionResult &result)
+	variable i, j, startIndex, endIndex
+	variable procCount = DimSize(procs, UTF_ROW)
+	variable funcCount = GetMaxFuncCount()
+	DFREF dfr = GetPackageFolder()
+	WAVE/WAVE funcLocations = UTF_Tracing#GetFuncLocations()
+	variable lbFuncList = FindDimLabel(funcLocations, UTF_COLUMN, "FUNCLIST")
+	variable lbFuncStart = FindDimLabel(funcLocations, UTF_COLUMN, "FUNCSTART")
+
+	Make/FREE=1/N=(procCount, funcCount)/T result.functions
+	Make/FREE=1/N=(procCount, funcCount) result.lines, result.calls, result.sums
+
+	for(i = 0; i < procCount; i++)
+		WAVE/T procFuncNames = funcLocations[i][lbFuncList]
+		WAVE procFuncLines = funcLocations[i][lbFuncStart]
+		funcCount = DimSize(procFuncNames, UTF_ROW)
+		result.count += funcCount
+
+		result.functions[i][] = procFuncNames[q]
```

Comment:
The indexing here leads to an RTE. You initialize the columns of result.functions with GetMaxFuncCount from all procedures, however procFuncNames from the current procedure can contain less than the maximum.

![image](https://user-images.githubusercontent.com/29093275/204051429-6aae8d1f-e711-40b5-aeee-fd2e39da77e8.png)

Reply:
Oh that was a big ouf. I tested this with two procedure files and a different number of functions now.